entity/User.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace User;
+
+use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Component\Validator\Mapping\ClassMetadata;
+use DateTime;
+
+class User
+{
+    private int $id;
+    private string $name;
+    private string  $email;
+    private string $password;
+    private DateTime $createdAt;
+
+    public function __construct(int $id, string $name, string $email, string $password)
+    {
+        $this->id = $id;
+        $this->name = $name;
+        $this->email = $email;
+        $this->password = $password;
+        $this->createdAt = new DateTime();
+    }
+    public static function loadValidatorMetadata(ClassMetadata $metadata)
+    {
+        $metadata->addPropertyConstraint('id', new Assert\GreaterThan([
+            'value' => 0,
+        ]));
+        $metadata->addPropertyConstraint('name', new Assert\NotBlank());
+        $metadata->addPropertyConstraint('name', new Assert\Length(['min' => 3]));
```

Comment:
Лучше сразу задавать все правила в одном месте:
```suggestion
        $metadata->addPropertyConstraint('name', [
            new Assert\NotBlank()
            new Assert\Length(['min' => 3]),
        ]);
```